cclib/parser/gaussianparser.py

```diff
@@ -1533,27 +1521,65 @@ def extract(self, inputfile, line):
                         orbital = line[11:20].strip()
                         self.aonames.append("%s_%s" % (atomname, orbital))
                         atombasis.append(i)
-
                     part = line[21:].replace("D", "E").rstrip()
                     temp = []
-
                     for j in range(0, len(part), 10):
                         temp.append(float(part[j:j+10]))
-
-                    nocoeffs[base:base + len(part) // 10, i] = temp
-
+                    coeffs[base:base + len(part) // 10, i] = temp
                 # Do the last update of atombasis.
                 if base == 0:
                     self.atombasis.append(atombasis)
-
                 # We now have aonames, so no need to continue.
                 if self.popregular:
                     break
-
+            return occnos,coeffs
+        if line[5:33] == "Natural Orbital Coefficients":
+            self.aonames = []
+            self.atombasis = []
+            self.nocoeffs = numpy.zeros((self.nmo, self.nbasis), "d")
+            self.nooccnos = []
+            self.popregular = False
```

Comment:
This is a good point. I actually removed the popregular because it isn't possible to have a NO calculation with popregular. I think this was lifted from the very similar MO parsing code? But I also used `set_attribute` and `append_attribute`. That was very useful thanks!